features/nanostack/FEATURE_NANOSTACK/targets/TARGET_SL_RAIL/NanostackRfPhyEfr32.cpp

```diff
@@ -325,21 +438,24 @@ static int8_t rf_start_cca(uint8_t *data_ptr, uint16_t data_length, uint8_t tx_h
             txOpt.waitForAck = false;
         }
 
-        //tr_debug("Called TX, len %d, chan %d, ack %d\n", data_length, channel, waiting_for_ack ? 1 : 0);
+        tr_debug("rf_start_cca: Called TX, len %d, chan %d, ack %d\n", data_length, channel, waiting_for_ack ? 1 : 0);
```

Comment:
Way too much debugging... `rf_start_cca()` is called for EVERY transmitted packet.

CCA process should be short, on some drivers it takes 8 symbols plus some processing, so for example 180 us. This print, on the other hand, takes 434 us in baud rate 115200.

This might work, but I might radically affect the device performance.

Reply:
All this debugging code we're removing is super useful for working on the driver, I keep having to re-add messages in my local copy to see what's happening.

Would you be open to wrapping them in a `SL_RADIO_DEBUG` macro or something instead of flat out removing so that they are excluded from normal builds but were easy to include when required?

It also kinda seems like this should be covered by the trace module filters? Though idk if it's possible to set a module to filter by default.

Reply:
I agree with you.
I would prefer SL_RADIO_DEBUG macro, because tracing macros are not yet supporting this filtering. By default, they only filter by level. In future, the filter-by-component might be added, but its not there yet.